api/src/test/java/org/openmrs/module/fhir/api/diagnosticreport/handler/LaboratoryHandlerTest.java

```diff
@@ -0,0 +1,268 @@
+        patient.setDeathDate(new Date());
+        patient.setCauseOfDeath(new Concept());
+        patient.setGender("male");
+        
+        List<PatientIdentifierType> patientIdTypes = patientService.getAllPatientIdentifierTypes();
+        assertNotNull(patientIdTypes);
+        PatientIdentifier patientIdentifier = new PatientIdentifier();
+        patientIdentifier.setIdentifier("123-0");
+        patientIdentifier.setIdentifierType(patientIdTypes.get(0));
+        patientIdentifier.setLocation(new Location(1));
+        patientIdentifier.setPreferred(true);
+        
+        Set<PatientIdentifier> patientIdentifiers = new LinkedHashSet<>();
+        patientIdentifiers.add(patientIdentifier);
+        
+        patient.setIdentifiers(patientIdentifiers);
+        
+        patientService.savePatient(patient);
+
+        return patient;
+    }
+
+    /**
+     * Creates and returns Person, given name
+     * 
+     * @param name name of the person to be created
+     * @return Person object
+     */
+    private Person newPerson(String name) {
+		Person person = new Person();
+		Set<PersonName> personNames = new TreeSet<>();
+		PersonName personName = new PersonName();
+		personName.setFamilyName(name);
+		personNames.add(personName);
+        person.setNames(personNames);
+        person.setPersonDateCreated(new Date());
+		return person;
+	}
+
+    /**
+     * Creates and returns Encounter, given the patient's Uuid
+     * 
+     * @param patient_uuid Uuid of the patient 
+     * @return Encounter object
+     */
+    private Encounter createEncounter(String patient_uuid)
+    {
+		Encounter encounter = new Encounter();
+		encounter.setLocation(new Location(1));
+		encounter.setEncounterType(new EncounterType(1));
+		encounter.setEncounterDatetime(new Date());
+		encounter.setPatient(patientService.getPatientByUuid(patient_uuid));
+		
+		EncounterRole role = new EncounterRole();
+		role.setName("role");
+		role = Context.getEncounterService().saveEncounterRole(role);
+		
+		Provider provider = new Provider();
+		provider.setIdentifier("id1");
+		provider.setPerson(newPerson("name"));
+		provider = Context.getProviderService().saveProvider(provider);
+		
+		Provider provider2 = new Provider();
+		provider2.setIdentifier("id2");
+		provider2.setPerson(newPerson("name2"));
+		provider2 = Context.getProviderService().saveProvider(provider2);
+		
+		encounter.addProvider(role, provider);
+		encounter.addProvider(role, provider2);
+        
+        EncounterType encounterType = new EncounterType("test", "test encounter type");
+        Context.getEncounterService().saveEncounterType(encounterType);
+        encounter.setEncounterType(encounterType);
+
+        Context.getAdministrationService().setGlobalProperty("fhir.encounter.encounterRoleUuid", role.getUuid());
+
+		EncounterService es = Context.getEncounterService();
+		es.saveEncounter(encounter);
+
+        encounterService.saveEncounter(encounter);
```

Comment:
Would be nice having the above lines nicely indented.

Reply:
@Ruhanga Somehow the code looked alright in my editor, but sending the whole code a tab backward and then a backward, indicated that lines were changed and seems to have fixed the indentation. Could you please have a look at it again?